Add staging/maxxroot course

Tocard opened the pull request.

maxxroot committed:
init plot manager

Tocard commented on disk_copy.py:
```diff
@@ -37,6 +39,22 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
     bladebit_manager_logger.log(SUCCESS, 'updated plot {} with status {} and dest {}'.format(plot_name, status, dest))
 
 
+def scan_plot_for_copy(directory_path):
```
rien d'obligatoire, mais pense à mettre les type, je les ai mis partout, pour la cohérence on va le garder et ça sera plus simple pour toi à relire.

Tocard replied:
cette fonction à mon sens est à split en deux truc

1. une fonction qui va te retourner la liste de tes plots quand tu l'appelles
2. l'autre fonction qui à ta boucle de vie du programme.

Tocard commented on disk_copy.py:
```diff
@@ -37,6 +39,22 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+        if filename.endswith('.plot'):
+
+
+while True:
```
l'idée est la mais ça doit etre une variable qui peut s'arrêter.

Tocard commented on disk_copy.py:
```diff
@@ -37,6 +39,22 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+
+while True:
+    staging_dir = config['staging_directories']
+    for directory_path in staging_dir:
```
Tu peux directement loop sur la config

Tocard commented on disk_copy.py:
```diff
@@ -37,6 +39,22 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+while True:
+    staging_dir = config['staging_directories']
+    for directory_path in staging_dir:
+        if os.path.exists(directory_path) and os.path.isdir(directory_path):
```
Te fait pas chier à verifier si les path existe, on feras un loader dans la config qui arreteras le soft si un des path est pas bon avant de lancer quoi que ce soit

Tocard commented on disk_copy.py:
```diff
@@ -37,6 +39,22 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+    staging_dir = config['staging_directories']
+    for directory_path in staging_dir:
+        if os.path.exists(directory_path) and os.path.isdir(directory_path):
+            scan_plot_for_copy(directory_path)
```
la tu fait de la recursivité, et je doute que ce sois ce que tu veuilles faire

Tocard commented on disk_copy.py:
```diff
@@ -37,6 +39,22 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
 
 
+def scan_plot_for_copy(directory_path):
+    for filename in os.listdir(directory_path):
```
part du principe que ceci te retourne une liste et qu'il sera plus simple de retourner la / les listes de plots et d'en faire quelque choses que de les traiter tous de façon unitaire

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,29 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+                plot_name = filename
+                source = staging_dir
+
+                insert_new_plot(plot_name, source)
```
pourquoi passer par des variable si elle non pas d'autres interet que les param de la fonction ?

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,29 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
-    bladebit_manager_logger.log(WARNING, get_plot_status_by_name(plot_name))
-    update_plot_by_name(plot_name, '/mnt/hydras01', 'done')
-    bladebit_manager_logger.log(WARNING, get_plot_status_by_name(plot_name))
+    scan_plots()
```
c'est en dehors d'une boucle la, donc une seule exec

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,29 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
-    update_plot_by_name(plot_name, '/mnt/hydras01', 'done')
-    bladebit_manager_logger.log(WARNING, get_plot_status_by_name(plot_name))
+    scan_plots()
+    cur.execute("SELECT * FROM plots WHERE status IS NULL")
```
une fonction get_plot_without_status ?

maxxroot replied:
oui car si j'ai aucun statuts c'est que mon plot est bien présent dans ma BDD mais qu'il n'est ni en cours d'envoie ni envoyé

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,29 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+        plot_name = result[0]
+        source = result[1]
+        bladebit_manager_logger.log(WARNING, get_plot_by_name(plot_name))
+        for final_dir_plot in config['directories_to_plot']: # a partir d'ici je galère donc à prendre avec des pincette ce qui est écrit
```
c'est quoi la strategie ? tu copies sur le premier, puis le deuxieme puis le troisieme etc etc ?

maxxroot replied:
oui mais je vais le modifier pour prendre les X premiers plots que j'envoie sur les X premier HDD

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,29 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+        source = result[1]
+        bladebit_manager_logger.log(WARNING, get_plot_by_name(plot_name))
+        for final_dir_plot in config['directories_to_plot']: # a partir d'ici je galère donc à prendre avec des pincette ce qui est écrit
+            update_plot_by_name(plot_name, final_dir_plot, 'in_progess')
```
l'idée est la mais c'est plus complexe, si tu lances la copy, tu met in progress, mais si tu echoues, il faut le remettre dans le pool

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,29 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+            source_path = os.path.join(source, plot_name)
+            shutil.move(source_path, final_dir_plot)
+        bladebit_manager_logger.log(WARNING, get_plot_status_by_name(plot_name))
+        update_plot_by_name(plot_name, '/mnt/hydras01', 'done')
```
comment tu verifies si c'est bon ?

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,29 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+        for final_dir_plot in config['directories_to_plot']: # a partir d'ici je galère donc à prendre avec des pincette ce qui est écrit
+            update_plot_by_name(plot_name, final_dir_plot, 'in_progess')
+            source_path = os.path.join(source, plot_name)
+            shutil.move(source_path, final_dir_plot)
```
à vue de nez, je dirais que c'est une commande blocante, ça va pas te rendre la main tant que tu n'as pas terminé

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,33 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+    results = cur.fetchall()
+    dest_dir = config['directories_to_plot']
+    for result in results:
+        plot_name = result[0]
```
result c'est un tuple, donc tu peux faire plot_name, source = result

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,33 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+        plot_name = result[0]
+        source = result[1]
+        bladebit_manager_logger.log(WARNING, get_plot_by_name(plot_name))
+        current_dest = dest_dir.pop(0) if dest_dir else None
```
pourquoi un pop ? la je pense que la logique est broken je comprends pas trop ou tu vas avec dest dir pop et current dest

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,33 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+        current_dest = dest_dir.pop(0) if dest_dir else None
+        if current_dest:
+            update_plot_by_name(plot_name, current_dest, 'in_progess')
+            source_path = os.path.join(source, plot_name)
```
on vas pas utiliser ça mais plutot

```python
from pathlib import Path

Path('./binaries/bladebit_cuda')
```

c'est plus costaud et gere nativement windows/linux

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,33 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+            update_plot_by_name(plot_name, current_dest, 'in_progess')
+            source_path = os.path.join(source, plot_name)
+            dest_path = os.path.join(current_dest, plot_name)
+            shutil.move(source_path, dest_path)
```
shutil.move retourne le repertoire de destination une fois le moove terminé, ça doit etre la condition pour l'update en bdd

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +40,33 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+            source_path = os.path.join(source, plot_name)
+            dest_path = os.path.join(current_dest, plot_name)
+            shutil.move(source_path, dest_path)
+            bladebit_manager_logger.log(INFO, f"Moved {plot_name} from {source_path} to {dest_path}")
```
si ça te dérange pas, je prefere la syntaxte .format()

```python
            bladebit_manager_logger.log(INFO, 'Moved {} from {} to {}'.format(plot_name, source_path, dest_path)

```

maxxroot committed:
récupération des plots pour transfert et check espace disk

maxxroot committed:
récupération des plots pour transfert et check espace disk

maxxroot committed:
v1 fonctionnel 5 copy simultané max

maxxroot committed:
v1 fonctionnel 5 copy simultané max

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +43,57 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+                insert_new_plot(filename, staging_dir)
+
+
+def process_plots(destination):
```
le typage :)

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +43,57 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
     bladebit_manager_logger.log(SUCCESS, 'updated plot {} with status {} and dest {}'.format(plot_name, status, dest))
 
 
+def left_space_on_directories_to_plots():
```
-> bool

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +43,57 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
-    update_plot_by_name(plot_name, '/mnt/hydras01', 'done')
-    bladebit_manager_logger.log(WARNING, get_plot_status_by_name(plot_name))
+    dest_dir = config['directories_to_plot']
+    num_process = 5
```
est-ce qu'on veut vraiment gerer comme ça et pas par disque de sortie max ?

maxxroot committed:
fix affichage log
fix db_lock
fix reset status in_progress au boot

Tocard commented on disk_copy.py:
```diff
@@ -1,28 +1,36 @@
 cur.execute("CREATE TABLE IF NOT EXISTS plots(plot_name TEXT UNIQUE, source TEXT, dest TEXT, status TEXT)")
 
 
+def get_plot_by_name(plot_name: str):
```
il manque le type retour de la fonction

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +45,67 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+        scan_plots()
+        try:
+            cur.execute("SELECT * FROM plots WHERE status IS NULL LIMIT 1")
+            result = cur.fetchone()
```
à verifier mais je pense que ce sera toujours true ça

Tocard commented on disk_copy.py:
```diff
@@ -37,12 +45,67 @@ def update_plot_by_name(plot_name: str, dest: str, status: str):
+                plot_name, source, _, _ = result
+                bladebit_manager_logger.log(WARNING, get_plot_by_name(plot_name))
+                update_plot_by_name(plot_name, destination, 'in_progress')
+                source_path = os.path.join(source, plot_name)
```
pas de os.path.join, Path de pathlib

Tocard commented on main.py:
```diff
@@ -19,9 +19,9 @@ def plot_runner():
-    plot_runner_process = multiprocessing.Process(target=plot_runner)
-    plot_runner_process.start()
-    plot_runner_process.join()
+#    plot_runner_process = multiprocessing.Process(target=plot_runner)
```
par contre, ça dit pas etre commit ça ;)

Tocard commented on config/config.yml:
```diff
@@ -9,18 +9,13 @@ contract_key: xch1tqlg956zauwf8tsljrafg5mwmnksrcgtzl9h6c9e9zrj27rrg0kqjm3gl3
 compression_level: 5
```
idem, le fichier de conf devrait pas etre commit sauf si on ajoute des trucs

maxxroot committed:
fix inserrt_new_plot

Commit:
some cleanup

Commit:
fix circular import

Commit:
fix orhcestrator

Commit:
convert simple yaml config to class with type input

Commit:
switch multiprocessing to threading

Commit:
rewrite logging part to handle debug

Commit:
fix debug and init way to split log into file

Commit:
add TU for sqlite

Commit:
add coverage

maxxroot committed:
fiw windows path error

Commit:
add test for scan plots

Commit:
Merge branch 'add_staging/maxxroot_course' of github.com:CryptoLFDM/bladebit_wrapper into add_staging/maxxroot_course

# Conflicts:
#	sqlite.py

Commit:
some fix & improove test

Commit:
rewrite process_plot

Commit:
add config for concurent process

Commit:
bypass config loader check to str cast

Commit:
add double check on plot_name to ensure we are not making double copy

Commit:
rewrite disk_copy logique

Commit:
rewrite method to get first free destination
add test on get_first_free_destination
add status to_process to replace old NULL/none one

Commit:
add some logs

Commit:
fix move update plots

Commit:
fix move update plots

Commit:
use list of directories issue from free scace calculation instead of config to avoid copy on full disk

Commit:
fix directories list

Commit:
fix error code

Commit:
fix double print for disk info

Commit:
fix double print for disk info

Commit:
Merge branch 'add_staging/maxxroot_course' of github.com:CryptoLFDM/bladebit_wrapper into add_staging/maxxroot_course

Commit:
reduce log display

Commit:
reduce log display
